vm/src/obj/objsequence.rs

```diff
@@ -386,15 +347,39 @@ pub fn seq_ge(
     zelf: &dyn SimpleSeq,
     other: &dyn SimpleSeq,
 ) -> Result<bool, PyObjectRef> {
-    Ok(seq_gt(vm, zelf, other)? || seq_equal(vm, zelf, other)?)
+    for (a, b) in Iterator::zip(zelf.iter(), other.iter()) {
+        if vm.bool_gt(a.clone(), b.clone())? {
+            return Ok(true);
+        } else if !vm.bool_eq(a.clone(), b.clone())? {
+            return Ok(false);
+        }
+    }
+
+    if zelf.len() == other.len() {
+        Ok(true)
```

Comment:
gt and ge only have this value difference. any idea for refactoring?

Reply:
```rust
fn seq_lt_le(
    vm: &VirtualMachine,
    zelf: &dyn SimpleSeq,
    other: &dyn SimpleSeq,
    eq: bool
) -> Result<bool, PyObjectRef> {
    for (a, b) in Iterator::zip(zelf.iter(), other.iter()) {
        if vm.bool_lt(a.clone(), b.clone())? {
            return Ok(true);
        } else if !vm.bool_eq(a.clone(), b.clone())? {
            return Ok(false);
        }
    }

    if zelf.len() == other.len() {
        Ok(eq)
    } else {
        Ok(zelf.len() < other.len())
    }
}

pub fn seq_lt(
    vm: &VirtualMachine,
    zelf: &dyn SimpleSeq,
    other: &dyn SimpleSeq,
) -> Result<bool, PyObjectRef> {
    seq_lt_le(vm, zelf, other, false)
}
pub fn seq_le(
    vm: &VirtualMachine,
    zelf: &dyn SimpleSeq,
    other: &dyn SimpleSeq,
) -> Result<bool, PyObjectRef> {
    seq_lt_le(vm, zelf, other, true)
}
```
What about the code above?

Reply:
```rust
#[inline]
fn seq_lte(
    vm: &VirtualMachine,
    zelf: &dyn SimpleSeq,
    other: &dyn SimpleSeq,
    eq: bool
) -> Result<bool, PyObjectRef> {
   ...
}
```
Adding `#[inline]` will help the eq parameter to be evaluated in compile time.

Or maybe we even don't need 2 versions of them
```rust
#[inline]
fn seq_cmp(
    vm: &VirtualMachine,
    zelf: &dyn SimpleSeq,
    other: &dyn SimpleSeq,
    obj_cmp: Fn(PyObjectRef, PyObjectRef) -> PyResult<bool>
    len_cmp: Fn(usize, usize) -> bool) {
   ....
}
```